### PR TITLE
Add test script for file operations

### PR DESCRIPTION
**Pull request by colinyoyo26**
Simply test if the result of file system operations about mkdir, create file, hard link and symbolic link are correct.

### CODE REVIEW

**Comment by jserv** [comment]
The reference results should be appended into git commit message.

File: script/test.sh [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+mkdir -p test  
+sudo umount test 2>/dev/null && \
+sudo rmmod $SIMPLEFS_MOD 2>/dev/null
+sleep 1
+modinfo $SIMPLEFS_MOD || exit 1
+sudo insmod $SIMPLEFS_MOD && \
+dd if=/dev/zero of=$IMAGE bs=1M count=$IMAGESIZE && \
+./$MKFS $IMAGE && \
+sudo mount -t simplefs -o loop $IMAGE test
```
Comment (jserv) [inline script/test.sh]: Here is `mount`. But when would you execute `umount`?

**Comment by jserv** [comment]
You can add more test cases such as longer file/directory names.

**Comment by colinyoyo26** [comment]
New test case:
    1. Create file/directory/symlink with longer name than file system limitation
    2. Test if the content, which be wrote into file, is correct
    3. Write datas, which is larger than maximum file size, into file

Also `umount` at the end of the file

File: script/test.sh (hunk 1 above)
Comment (colinyoyo26) [reply script/test.sh]: I originally wanted to keep it there for manual testing, but it doesn't seem to make sense though.

**Comment by colinyoyo26** [comment]
In this amendation:
1. Suppress the messages from `dd` .
2. Mention write file test case in commit message.
3. Sleep 1 second before `umount`

File: script/test.sh [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+sleep 1
+sudo rmmod $SIMPLEFS_MOD 2>/dev/null
+sleep 1
+(modinfo $SIMPLEFS_MOD || exit 1) && \
+echo && \
+sudo insmod $SIMPLEFS_MOD  && \
+dd if=/dev/zero of=$IMAGE bs=1M count=$IMAGESIZE status=none&& \
+./$MKFS $IMAGE && \
+sudo mount -t simplefs -o loop $IMAGE test && \
+cd test
```
Comment (jserv) [inline script/test.sh]: You can make use of `pushd` and `popd`.

**Comment by jserv** [comment]
Improve the subject of git commit message.

File: script/test.sh (hunk 2 above)
Comment (colinyoyo26) [reply script/test.sh]: Thanks for review, I've replaced `cd` with `pushd` and `popd`.

**Comment by jserv** [comment]
Thank @colinyoyo26 for contributing!

**Pull request merged**